### PR TITLE
Document operation

### PR DESCRIPTION
**Pull request by patrick-werner**

### CODE REVIEW

**Comment by jamesagnew** [comment]
This looks awesome to me.

@GeorgeZywiel can you give it a look too?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/provider/BaseJpaResourceProviderCompositionDstu2.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * #L%
+ */
+
+public class BaseJpaResourceProviderCompositionDstu2 extends JpaResourceProviderDstu2<Composition> {
+
+	/**
+	 * Composition/123/$document
+	 */
+	//@formatter:off
+	@Operation(name = "document", idempotent = true, bundleType=BundleTypeEnum.SEARCHSET)
```
Comment (GeorgeZywiel) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/provider/BaseJpaResourceProviderCompositionDstu2.java]: For consistency, the name should be "$document",  and it should be added to / imported from `JpaConstants.java`

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/provider/dstu3/BaseJpaResourceProviderCompositionDstu3.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+public class BaseJpaResourceProviderCompositionDstu3 extends JpaResourceProviderDstu3<Composition> {
+
+	/**
+	 * Composition/123/$document
+	 *
+	 * @param theRequestDetails
+	 */
+	//@formatter:off
+	@Operation(name = "document", idempotent = true, bundleType=BundleTypeEnum.DOCUMENT)
```
Comment (GeorgeZywiel) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/provider/dstu3/BaseJpaResourceProviderCompositionDstu3.java]: same comment as Dstu2 version

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/provider/r4/BaseJpaResourceProviderCompositionR4.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+public class BaseJpaResourceProviderCompositionR4 extends JpaResourceProviderR4<Composition> {
+
+	/**
+	 * Composition/123/$document
+	 *
+	 * @param theRequestDetails
+	 */
+	//@formatter:off
+	@Operation(name = "document", idempotent = true, bundleType=BundleTypeEnum.DOCUMENT)
```
Comment (GeorgeZywiel) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/provider/r4/BaseJpaResourceProviderCompositionR4.java]: same comment as Dstu2 version

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4BundleTest.java [hunk 4]
```diff
@@ -35,7 +35,8 @@ public void testBundlePreservesFullUrl() {
 		IIdType id = myClient.create().resource(bundle).execute().getId();
```
Comment (GeorgeZywiel) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4BundleTest.java]: The parameter `myClient` should be renamed to `ourClient` to match the renamed parameter in `BaseResourceProviderR4Test.java`

(3 occurrences in this file)

**Pull request merged**